dexlib2/src/main/java/com/android/tools/smali/util/CharSequenceUtils.java

```diff
@@ -30,21 +30,28 @@
 
 package com.android.tools.smali.util;
 
-import com.google.common.base.Function;
-import com.google.common.base.Functions;
-import com.google.common.collect.Lists;
+import java.util.function.Function;
+import java.util.Iterator;
 
 import java.util.List;
 
 public class CharSequenceUtils {
-    private static final Function<Object, String> TO_STRING = Functions.toStringFunction();
+    private static final Function<Object, String> TO_STRING = new Function<Object,String>() {
+        @Override
+        public String apply(Object o) {
+            if (o == null) {
+                throw new NullPointerException();
```

Comment:
Any particular reason for the `null` check?  The `o.toString()` will NPE anyway. Otherwise consider `Objects.requireNonNull` (available from Java 7), which also have an optional second message parameter.

Reply:
Nothing particular, just copied over from the guava [toStringFunction](https://github.com/google/guava/blob/1a583c32fadcdd5f0ce851d80f8d8613ea426778/guava/src/com/google/common/base/Functions.java#L71), but you're right that it'll NPE anyways. Removing.